docs/upgrade.html

```diff
@@ -27,6 +27,14 @@ <h5><a id="upgrade_270_notable" href="#upgrade_270_notable">Notable changes in 2
         <code>default.api.timeout.ms</code>, and Kafka Streams' new <code>task.timeout.ms</code> parameters instead.
         Note that parameter <code>retry.backoff.ms</code> is not impacted by this change.
     </li>
+    <li>Altering non-reconfigurable configs of existent listeners causes <code>InvalidRequestException</code>.
+        By contrast, the previous behavior would have caused the updated configuration to be persisted, but it wouldn't
+        take effect until the broker was restarted. This change breaks behavior compatibility but the old behavior is not
+        worth keeping since that behavior is unintended. see
+        <a href="https://github.com/apache/kafka/pull/9284">KAFKA-10479</a> for more discussion.
+        see <code>DynamicBrokerConfig.DynamicSecurityConfigs</code> and <code>SocketServer.ListenerReconfigurableConfigs</code>
+        for reconfigurable configs of existent listeners.
+    </li>
```

Comment:
Does it really break compatibility? It seems like the previous behavior wasn't working as intended anyway.

Reply:
> Does it really break compatibility? It seems like the previous behavior wasn't working as intended anyway.

It seems to me the APIs behavior is changed so the compatibility is broken.

For example, the code which is used to change non-reconfigurable configs gets exception now so users have to write something to handle "new" exception. Of course, that case should be very rare so mentioning this in docs/upgrade.html is enough to this incompatible change.

Reply:
I know what you're saying, but this change aligns the implementation with the specified behavior. The previous behavior was a bug as it failed to report an error even though it did not change the config dynamically.

Reply:
> The previous behavior was a bug as it failed to report an error even though it did not change the config dynamically.

ok. it makes sense. I will remove this comment from ```docs/upgrade.html```

Reply:
I think we should keep the comment, I was suggesting we tweak the description.

Reply:
> I think we should keep the comment, I was suggesting we tweak the description.

got it.